### PR TITLE
Speed up OSM processing and switch to Python

### LINKED ISSUE #7: OSM-POI: Building footprint for relations
So far we're only generating building footprints for OSM objects of type `way`.

`Relations` also need to be supported. Special attention here deserves the case of multi polygons. Some may have holes and others might be separate areas that semantically belong together (e.g., a university complex consisting of several buildings).

The building footprints follow the [GeoJSON standard](https://de.wikipedia.org/wiki/GeoJSON).

The relevant function is in `kuwala-pipelines/osm-poi/src/data/processor/pbfParser/geoParser/coordsParser.js` called *`convertMultiPolygon`*.

The members can be retrieved from the level db which temporarily saves nodes used for building footprints (see *`convertToGeoJSONCoords`*).

### LINKED ISSUE #48: Core: Use Neo4j Spark connector for importing data
Instead of sending the insertion queries through the Neo4j Python driver, we should use the Spark Neo4j connector to read the preprocessed population and OSM Parquet files and make the inserts. The Spark driver batches the queries automatically, which should speed up the entire population process.

It depends on #47

### LINKED ISSUE #49: OSM-POI: Switch to Python and use Parquet + Spark
To use Spark for the transformations, we switch to Python for this pipeline.

We can use this [repo](https://github.com/adrianulbona/osm-parquetizer) to parse the OSM pbf files to Parquet files.

### PR DESCRIPTION
**Pull request by mattigrthr**
Closes #7 
Closes #48 
Closes #49

### CODE REVIEW

File: kuwala/core/neo4j/importer/src/Neo4jConnection.py [hunk 1]
```diff
@@ -46,39 +44,26 @@ def query_graph(q, parameters=None, db=None):
...
-        if len(batch) == batch_size:
-            send_query(batch)
-
-            batch = list()
-
-    if len(batch) > 0:
-        send_query(batch)
-
-    close_connection()
+def spark_send_query(df, query):
```
Comment (bob-bn) [inline kuwala/core/neo4j/importer/src/Neo4jConnection.py:47]: I would change the name of the function here to describe exactly wat it does... write_df_to_neo4j_with_override(..)

File: kuwala/core/neo4j/importer/src/PipelineConnector.py [hunk 2]
```diff
@@ -33,40 +42,76 @@ def connect_h3_indexes():
...
+    '''
+
+    Neo4jConnection.spark_send_query(df, query)
+
 
 # Create one single POI node combining OSM and Google
 def connect_pois(df_osm: DataFrame, df_google: DataFrame):
-    df_osm = df_osm.select('id', 'h3Index').withColumnRenamed('h3Index', 'h3IndexOsm').withColumnRenamed('id', 'osmId')
+    start_time = time.time()
+    df_osm = df_osm.select('id', 'h3_index') \
```
Comment (bob-bn) [inline kuwala/core/neo4j/importer/src/PipelineConnector.py:86]: [suggestion] just for later, all these hardcoded name mappings are better to be moved to a KEY / Value dictionary that is importable across the board. this way you avoid potential mistyping...but nothing to action now.

File: kuwala/core/neo4j/importer/src/PipelineConnector.py [hunk 3]
```diff
@@ -33,40 +42,76 @@ def connect_h3_indexes():
...
         .withColumn('osmId', df_google['osmId'].cast(LongType())) \
         .withColumn('osmId', concat('type', 'osmId')) \
         .select('id', 'osmId', 'confidence', 'h3Index') \
         .withColumnRenamed('id', 'googleId') \
         .withColumnRenamed('h3Index', 'h3IndexGoogle')
-    df_pois = df_osm.join(df_google, on=['osmId'], how='left')
 
-    Neo4jConnection.connect_to_graph()
-    Neo4jConnection.query_graph('CREATE CONSTRAINT poi IF NOT EXISTS ON (p:Poi) ASSERT p.id IS UNIQUE')
+    @udf(returnType=StringType())
```
Comment (bob-bn) [inline kuwala/core/neo4j/importer/src/PipelineConnector.py:96]: [suggestion] UDFs should be importable functions I am sure there might be uses cases that the same peice of code will be helpful.

File: kuwala/core/neo4j/importer/src/PipelineConnector.py [hunk 4]
```diff
@@ -33,40 +42,76 @@ def connect_h3_indexes():
...
+    def get_poi_id(confidence, h3_index_google, h3_index_osm, osm_id):
```
Comment (bob-bn) [inline kuwala/core/neo4j/importer/src/PipelineConnector.py:97]: Better naming here could be more future proof, get_poi_id does not tell me much about what this function does. build_poi_id_based_on_confidence ?

**Review by bob-bn — APPROVED** [review]

File: kuwala/core/neo4j/importer/src/PipelineConnector.py (hunk 2 above)
Comment (mattigrthr) [reply kuwala/core/neo4j/importer/src/PipelineConnector.py:86]: Makes sense. I opened an issue for that.

File: kuwala/core/neo4j/importer/src/PipelineConnector.py (hunk 3 above)
Comment (mattigrthr) [reply kuwala/core/neo4j/importer/src/PipelineConnector.py]: Agree, I added a new commit where I included general Spark UDFs to `python_utils`.

**Review by bob-bn — APPROVED** [review]

**Pull request merged**